eng/targets/ResolveReferences.targets

```diff
@@ -157,6 +157,76 @@
 
       <!-- Remove reference items that have been resolved to a LatestPackageReference item. -->
       <Reference Remove="@(_LatestPackageReferenceWithVersion)" />
+    </ItemGroup>
+
+    <!-- Substitute references to Extensions packages with with ref assemblies -->
+    <ItemGroup Condition="'$(ReferenceReferenceAssemblies)' == 'true' AND '$(TargetFramework)' == 'netcoreapp3.0' AND '$(DisableExtensionsRefAssembliesSwap)' != 'true' ">
+      <!-- References that have ref assemblies in Microsoft.Extensions.Internal.Refs -->
```

Comment:
I created this list based on what I found in the ref pack. This list is very similar to the list at https://github.com/aspnet/AspNetCore/blob/f676c249d25eb438ffc282edb551d86eea1d9709/eng/SharedFramework.External.props#L10-L56 but there are some additional assemblies in the ref pack.  For example Microsoft.Extensions.Logging.AzureAppServices which is used in https://github.com/aspnet/AspNetCore/blob/f676c249d25eb438ffc282edb551d86eea1d9709/src/Azure/AzureAppServicesIntegration/src/Microsoft.AspNetCore.AzureAppServicesIntegration.csproj#L15. 

However, curating this list here does seem unmaintainable and wrong so I'm considering using ExternalAspNetCoreAppReference instead. Especially due to the fact that most of the "missing" references in that item group are only used by samples in this repo which will bypass this logic anyway.

Reply:
I'll look into alternative ways of generating this list so consider it a quick and dirty way to prove out the concept the rest of the mechanism is sound. Will remove this list once there's positive results.